UofT-Frosh-Orientation/orientation-website
# 498 Ability to delete accounts

This PR is about adding a feature to delete user accounts in the page "frosh-info-table":

- [x] Add new scope for deleting accounts to frontend
- [x] Add new account delete route (with user id)
- [x] Add delete column to frosh data table page with X icon and functionality to frontend (and if correct auth scopes)
- [x] Add delete confirmation popup on frontend before proceeding

To test this you will need to have the scopes to view and delete info in the frosh-info-table (`"accounts:delete", "accounts:edit", "accounts:read", "admin:all", "froshData:unRegisteredUsers", "froshGroupData:all"`). I don't actually know if you need all these scopes but I had them when I tested it myself.

Steps to test:
- Add the following auth-scopes to your user: `"accounts:delete", "accounts:edit", "accounts:read", "admin:all", "froshData:unRegisteredUsers", "froshGroupData:all"`
- Go to `http://localhost:3000/frosh-info-table`
- If you can already see some users, try to delete one of them by pressing the 'X' button in the "Delete Accounts" column
- If you don't see any users or only see one, create more users first before trying to delete
- Try to test deleting both leedur, frosh, and unregistered user accounts
- Check if the user is deleted in the database :)
